### ISSUE TITLE
Allow extensions to create file based output channel 

### ISSUE THREAD

**Issue by sandy081** [Microsoft/vscode#41672]
Allow extensions to create file based output channel

**Comment by TomasHubelbauer** [comment]
@sandy081 Can you clarify what this would actually allow the extension authors to do? Something like treating output channels as files where changes to the file would get reflected in the output channel contents (allowing us to rewrite already output content)?

**Comment by sandy081** [comment]
@TomasHubelbauer Updated the description.

**Comment by TomasHubelbauer** [comment]
@sandy081 is it feasible for this to be developed by an external contributor? Is it easy enough? To me it sounds like a file system watcher on the file and some hooks the the Output channels APIs could do the job. If I'm correct, would you mind posting some pointers in the code the community could use to try and pick up this work?

**Comment by sandy081** [comment]
@TomasHubelbauer Yes this is easy to implement. We already have file based output channels internally. This is only to expose it as an API. 

Since it is an API we have some protocols to follow, we would need extension authors requiring for this feature with proper use cases. Since there are no real time requirements from extension authors, I would keep this on hold until we get some.

**Comment by TomasHubelbauer** [comment]
Please check out https://github.com/rust-lang-nursery/rls-vscode/pull/306#issuecomment-387062340 for one possible use case.
